filebeat/input/filestream/identifier.go

```diff
@@ -87,27 +87,33 @@ func newFileIdentifier(ns *common.ConfigNamespace) (fileIdentifier, error) {
 		return nil, fmt.Errorf("no such file_identity generator: %s", identifierType)
 	}
 
-	return f(ns.Config())
+	return f(ns.Config(), suffix)
 }
 
 type inodeDeviceIdentifier struct {
-	name string
+	name   string
+	suffix string
```

Comment:
It looks like each strategy now needs to implement support for the `suffix`. Alternatively consider to generalize the suffix support via:

```
type suffixIdentifier struct {
  inner identifier
  suffix string
}

func withSuffix(inner identifier, suffix string) identifier {
  if suffix == "" {
    return inner
  }
  return suffixIdentifier{inner: inner, suffix: suffix}
}

func (s suffixIdentifier) GetSource(e loginp.FSEvent) fileSource {
  fs := s.inner.GetSource(e)
  fs.name += "-" + s.suffix
  return fs
}
```